### ISSUE TITLE
Get rid of references and code specific to Python 2.4

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5387]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)| @jesstess reported|
|-|-|
|Trac ID|trac#5387|
|Type|enhancement|
|Created|2011-11-13 21:41:57Z|
|Branch|https://github.com/twisted/twisted/tree/remove-py24-refs-5387-4|

Once we drop support for 2.4 in ticket [ticket:4962 4962] and close [#5385](https://github.com/twisted/twisted/issues/5385) (py2.2) and [#5386](https://github.com/twisted/twisted/issues/5386) (py2.3). Some of these comments can just get removed. Some might require removal of code as well.

```
twisted/conch/test/test_filetransfer.py:        # In Python 2.4, the bad import has already been cleaned up for us.
twisted/internet/defer.py:    WARNING: this function will not work in Python 2.4 and earlier!
twisted/internet/defer.py:    generators.  If you need to be compatible with Python 2.4 or before, use
twisted/internet/unix.py:            # Abstract namespace sockets aren't well supported on Python 2.4.
twisted/python/dist.py:    older versions (currently the only supported older version is 2.4), checks
twisted/python/filepath.py:    Provide random data in versions of Python prior to 2.4.  This is an
twisted/python/filepath.py:            # Under Python 2.4 on Windows, WindowsError only has an errno
twisted/python/filepath.py:            # in Python 2.4 will result in a Windows error code of
twisted/python/reflect.py:                if (execName is None or # python 2.4+, post-cleanup
twisted/python/test/test_dist.py:    Tests for L{_checkCPython} when used on a Python 2.4-like platform, when
twisted/python/test/test_zipstream.py:        # In python 2.4 and earlier, consistency between the directory and the
twisted/python/util.py:    This is mostly necessary in Python 2.4 which implements L{id} to sometimes
twisted/test/test_defgen.py:## syntax error in Python 2.4 and before.
twisted/trial/runner.py:            # objects do not have nice looking string formats on Python 2.4.
twisted/trial/test/mockdoctest.py:# to test trial's doctest support with python2.4
twisted/trial/unittest.py:        # in __call__, whereas 2.4 defines the code in run.
twisted/trial/util.py:        else: # use hotshot, profile is broken in 2.4
twisted/trial/util.py:                    # Python 2.4 has fixed this.  Yay!
twisted/web/client.py:    # Python 2.4 doesn't have these symbolic constants
twisted/web/test/test_xmlrpc.py:    Tests for the C{useDateTime} flag on Python 2.4.
twisted/web/xmlrpc.py:# On Python 2.4 and earlier, DateTime.decode returns unicode.
```

Attachments:

* [ticket-5387.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3b4/3b442a3870e6ab894b40142a03c4dcba09b1569e/26c529492eccb6beb0ee260ad406b413c45d85f6.diff) (8993 bytes) - added by bartekci on 2012-03-13 21:24:17Z - Initial work for review. Partially done deprecation
* [5387.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3b4/3b442a3870e6ab894b40142a03c4dcba09b1569e/2e70501f9288c070c8936f572c1c30e014d51591.patch) (13350 bytes) - added by bartekci on 2012-03-14 02:27:12Z - Added complete changes for ticket and news files. A few items referenced in the original ticket were left unchanged as they seemed rellevant to the codebase past Python 2.4
* [5387.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3b4/3b442a3870e6ab894b40142a03c4dcba09b1569e/2ebc6d084f17a3d4bd9da26a0f8b93a3ef8073bc.patch) (12769 bytes) - added by bartekci on 2012-03-15 15:53:24Z - Updated patch with newsfile typo fixes and additional doc changes as requested.

<details><summary>Searchable metadata</summary>

```
trac-id__5387 5387
type__enhancement enhancement
reporter__jesstess jesstess
priority__normal normal
milestone__ 
branch__branches_remove_py24_refs_5387_4 branches/remove-py24-refs-5387-4
branch_author__bartekci__MostAwesomeDude__thijs bartekci, MostAwesomeDude, thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1321220517000000 1321220517000000
changetime__1377282466000000 1377282466000000
version__None None
owner__thijs thijs
cc__jesstess cc__bartekci cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

Also "Programming Language :: Python :: 2.4" in setup.py

**Comment by twisted-trac** [comment]
|<img alt="bartekci's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|bartekci set owner to bartekci<br>bartekci set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="bartekci's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|bartekci commented|
|-|-|

Initial work, still in progress, but would be happy if someone reviewed it to make sure it's appropriately done. Here's a summary of changes done so far:


twisted/web/test/test_xmlrpc.py
-- Removed Python 2.4 specific test and flags.

twisted/web/xmlrpc.py
-- Removed < Python 2.5 check in __setattr__
-- Removed hack for DateTime.decode

twisted/web/client.py
-- Adjusted getattr(os, 'SEEK_SET', 0) to os.SEEK_SET, kept private vclass variable though. Necessary?

twisted/trial/util.py
-- Removed 2.4 hack under profiled
-- DIDNOTFIX: Python 2.4 has fixed this, line 286. del sys.modules try/catch
l

twisted/trial/unittest.py
-- Removed overrides to __call__ and run. Python 2.5/6 handle this as
implemented.

twisted/trial/test/mockdoctest.py
-- Removed reference to python 2.4 in doc, but test seems valid.

twisted/trial/runner.py
-- Removed deprecated DocTestCase.

twisted/test/test_defgen.py
-- Removed string exec hack for InlineCallbacksTests

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess removed owner<br>@jesstess set status to `new`|
|-|-|

Thanks for this initial work, bartekci! If you want your work reviewed, please add the 'review' keyword and assign the ticket to <blank>. This will make it show up on http://twistedmatrix.com/trac/report/11 and in the review reminders on IRC. Otherwise it's likely to get lost in the shuffle. :) I'll go ahead and set it now.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph commented|
|-|-|

(Just updating markup, please ignore.)

**Comment by twisted-trac** [comment]
|[<img alt="MostAwesomeDude's avatar" src="https://avatars.githubusercontent.com/u/118035?s=50" width="50" height="50">](https://github.com/MostAwesomeDude)<a name="note_7"></a>|@MostAwesomeDude commented|
|-|-|

(In [33806]) Branching to 'remove-py24-refs-5387'

**Comment by twisted-trac** [comment]
|[<img alt="MostAwesomeDude's avatar" src="https://avatars.githubusercontent.com/u/118035?s=50" width="50" height="50">](https://github.com/MostAwesomeDude)<a name="note_8"></a>|@MostAwesomeDude commented|
|-|-|

(In [33807]) Applying patch from [[#5387](https://github.com/twisted/twisted/issues/5387)](https://github.com/twisted/twisted/issues/5387).

Refs #5387

**Comment by twisted-trac** [comment]
|[<img alt="MostAwesomeDude's avatar" src="https://avatars.githubusercontent.com/u/118035?s=50" width="50" height="50">](https://github.com/MostAwesomeDude)<a name="note_9"></a>|@MostAwesomeDude commented|
|-|-|

(In [33808]) t.p.util: Fix trailing whitespace.

Refs [#5387](https://github.com/twisted/twisted/issues/5387)

**Comment by twisted-trac** [comment]
|[<img alt="MostAwesomeDude's avatar" src="https://avatars.githubusercontent.com/u/118035?s=50" width="50" height="50">](https://github.com/MostAwesomeDude)<a name="note_10"></a>|@MostAwesomeDude commented|
|-|-|

Looks like I've added myself to the authors list. Whatever.

This largely looks decent. The topfiles are **not** okay. I don't know what "Tornadio" is, but I'm pretty sure you meant something else. There are also spelling mistakes ("byres") which should be corrected.

I'm going to kick off a build for this branch; you can follow along at home at http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-py24-refs-5387

I know I'm forfeiting points, but I'm going to leave on the review keyword to attract other eyes. I don't see any code problems, but this is a very wide-reaching patch.

**Comment by twisted-trac** [comment]
|[<img alt="jerith's avatar" src="https://avatars.githubusercontent.com/u/43731?s=50" width="50" height="50">](https://github.com/jerith)<a name="note_11"></a>|@jerith commented|
|-|-|

I only have some minor stylistic comments.

 * In `python/dist.py` removing the Python 2.4 comment in the docstring leaves a partial line. I'd prefer the paragraph be reflowed to look tidier.
 * In `internet/defer.py` all references to Python 2.4 have been removed, but the reference to "Python 2.5 generators" is still there. Since we no longer support any version of Python /without/ generators, I'd drop the version from that sentence.
 * In `trial/test/mockdoctest.py` there is a reference to a `__test__` attribute that doesn't seem to exist. This isn't strictly part of this ticket, but I think it can be cleaned up as well.

The code changes look okay. There's only one test failure on my system (some XPath thing in twisted.words), and that looks unrelated to the changes in this branch. (The same test fails in trunk.)

```
Ran 7295 tests in 121.716s
FAILED (skips=1468, expectedFailures=11, failures=1, successes=5815)
```

**Comment by twisted-trac** [comment]
|[<img alt="jerith's avatar" src="https://avatars.githubusercontent.com/u/43731?s=50" width="50" height="50">](https://github.com/jerith)|@jerith set owner to @MostAwesomeDude|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_13"></a>|@ralphm commented|
|-|-|

There is also some version-specific code (for before Python 2.3.2) in `twisted.words.protocols.jabber.xmpp_stringprep`.

Also, I think your failing test is in `test_sip`, not xish.

**Comment by twisted-trac** [comment]
|<img alt="rockstar's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|rockstar set owner to rockstar|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="MostAwesomeDude's avatar" src="https://avatars.githubusercontent.com/u/118035?s=50" width="50" height="50">](https://github.com/MostAwesomeDude)<a name="note_16"></a>|@MostAwesomeDude commented|
|-|-|

(In [33851]) Update from 5387.2.patch.

Refs [#5387](https://github.com/twisted/twisted/issues/5387)

**Comment by twisted-trac** [comment]
|[<img alt="MostAwesomeDude's avatar" src="https://avatars.githubusercontent.com/u/118035?s=50" width="50" height="50">](https://github.com/MostAwesomeDude)<a name="note_17"></a>|@MostAwesomeDude commented|
|-|-|

Updated the branch to the latest stuff from the patch. Please review.

bartekci, thank you for your hard work on this issue.

**Comment by twisted-trac** [comment]
|<img alt="rockstar's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_18"></a>|rockstar removed owner|
|-|-|

Hi bartekci, thanks for this patch.

Since you've removed DocTestCase here, you should probably also assign [#5554](https://github.com/twisted/twisted/issues/5554) to yourself and close it when this ticket makes it to trunk.

After talking with dreid, I think it's probably better to ask that we hold off on completion of this ticket until after we close [#5385](https://github.com/twisted/twisted/issues/5385) and [#5386](https://github.com/twisted/twisted/issues/5386).  A large reason is that there are many things like in t.internet.unix._UNIXPort that do things like "if sys.version_info > (2, 5)", and so we should probably remove that support in this ticket, *but* we should make sure that everything older than 2.4 is also taken care of.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_22"></a>|@thijstriemstra commented|
|-|-|

(In [37918]) remove more python2.4 references, add news files. refs [#5387](https://github.com/twisted/twisted/issues/5387)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_23"></a>|@thijstriemstra commented|
|-|-|

(In [37919]) update setup files, refs [#5387](https://github.com/twisted/twisted/issues/5387)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_24"></a>|@thijstriemstra commented|
|-|-|

(In [37920]) Branching to 'remove-py24-refs-5387-3'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_25"></a>|@thijstriemstra commented|
|-|-|

(In [37921]) merge forward, refs [#5387](https://github.com/twisted/twisted/issues/5387)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_26"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Forced a [build](https://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-py24-refs-5387-3), up for review.

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_27"></a>|@Julian commented|
|-|-|

Should the reference & conditional rockstar mentioned in t.i.unix._UNIXPort now be removed?

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)|@tomprince set owner to @tomprince<br>@tomprince set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_29"></a>|@tomprince set owner to @thijstriemstra<br>@tomprince set status to `new`|
|-|-|

1.`twisted/conch/topfiles/setup.py`: Do we even need the conditional anymore? If the only versions we support accept classifiers, we might as well always provide them.
   * This is in fact true of every `setup.py`.
 2. `twisted.internet.defer`: Don't delete the example.
 3. `twisted.python.filepath`
   1. Why are you removing the reference to randomBytes? If it is for compatibility, it should be deprecated in a separate ticket.
   2. The code in `children` can be simplified (but perhaps this should entirely be a separate ticket:
     * `.winerror` is always present now.
     * The comment suggests that it might be possible to simply use the posix error code on windows as well.
     * Both of those changes probably require test changes.
 4. `twisted/python/test/test_dist.py`:
   * I think the comment about being tests for a 2.4-like platform are still accurate.
   * I wonder, though, if we should be claiming that python is cpython in the cases described by the tests. None of 2.5, 2.6 or 2.7 fall into this case, so anything that we might conceivably support that behaves this way can't be cpython.
 5. `twisted/trial/runner.py`:
   * The comment you are deleting suggests that we could simplify the code now. If this is the case, we should leave some evidence of this (new ticket?).
 6. `twisted/web/client.py`:
   * Since these constans are always present, we can simply use them directly, rather than the aliases you are touching.
 7. `twisted/web/xmlrpc.py`:
   1. The `__setattr__` is now superfluous.
   2. (minor) `L{xmlrpclib}` instead of `C{xmlrpclib}`.
 8. There appears to be an error on the `py-select-gc` builder.
 9. I haven't looked at coverage of the changed code, but [this](http://buildbot.twistedmatrix.com/builders/pyflakes/builds/239/steps/pyflakes/logs/new%20pyflakes%20errors) makes me thing that at least some of it isn't.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_30"></a>|@thijstriemstra commented|
|-|-|

(In [38650]) address review comments, refs [#5387](https://github.com/twisted/twisted/issues/5387)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_31"></a>|@thijstriemstra commented|
|-|-|

(In [38652]) Branching to 'remove-py24-refs-5387-4'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_32"></a>|@thijstriemstra commented|
|-|-|

(In [38653]) merge forward, resolve conflicts. refs [#5387](https://github.com/twisted/twisted/issues/5387)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_33"></a>|@thijstriemstra removed owner|
|-|-|

Thanks for your review!

Replying to [tom.prince](#note_29):
>  1. `twisted/conch/topfiles/setup.py`: Do we even need the conditional anymore? If the only versions we support accept classifiers, we might as well always provide them.
>    * This is in fact true of every `setup.py`.

Removed the conditionals in setup.py's.

>  2. `twisted.internet.defer`: Don't delete the example.

Fixed.

>  3. `twisted.python.filepath`
>    1. Why are you removing the reference to randomBytes? If it is for compatibility, it should be deprecated in a separate ticket.

This was part of a patch supplied by someone else; restored the reference again.

>    2. The code in `children` can be simplified (but perhaps this should entirely be a separate ticket:
>      * `.winerror` is always present now.
>      * The comment suggests that it might be possible to simply use the posix error code on windows as well.
>      * Both of those changes probably require test changes.

I tried simplifying it but after several tries on the windows buildslave this is what worked. I'd prefer to handle this in a different ticket, i dont have access to a local windows machine for proper testing etc.

>  4. `twisted/python/test/test_dist.py`:
>    * I think the comment about being tests for a 2.4-like platform are still accurate.

Restored the comment.

>    * I wonder, though, if we should be claiming that python is cpython in the cases described by the tests. None of 2.5, 2.6 or 2.7 fall into this case, so anything that we might conceivably support that behaves this way can't be cpython.

What do you suggest, should we rename `_checkCPythonWithEmptyPlatform` to `_checkPythonWithEmptyPlatform`? And/or remove the `test_cpython` testcase?

>  5. `twisted/trial/runner.py`:
>    * The comment you are deleting suggests that we could simplify the code now. If this is the case, we should leave some evidence of this (new ticket?).

Yes, I simplified it and it still passes the test. I think opening a new ticket will only create more work since its already covered by a test.. thoughts?

>  6. `twisted/web/client.py`:
>    * Since these constans are always present, we can simply use them directly, rather than the aliases you are touching.

Fixed.

>  7. `twisted/web/xmlrpc.py`:
>    1. The `__setattr__` is now superfluous.
>    2. (minor) `L{xmlrpclib}` instead of `C{xmlrpclib}`.

Fixed.

>  8. There appears to be an error on the `py-select-gc` builder.

Forced a new [build](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-py24-refs-5387-4)

>  9. I haven't looked at coverage of the changed code, but [this](http://buildbot.twistedmatrix.com/builders/pyflakes/builds/239/steps/pyflakes/logs/new%20pyflakes%20errors) makes me thing that at least some of it isn't.

The coverage report i generated shows (the changes made to) `Proxy` is covered.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_34"></a>|@thijstriemstra commented|
|-|-|

Replying to [Julian](#note_27):
> Should the reference & conditional rockstar mentioned in t.i.unix._UNIXPort now be removed?

I think that should be handled in [#5715](https://github.com/twisted/twisted/issues/5715).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_35"></a>|@tomprince set owner to @thijstriemstra|
|-|-|

1.`twisted/trial/runner.py`: `"%s" % (repr(x),)` is the same as `"%r" % (x,)`
 2. ``
>> I wonder, though, if we should be claiming that python is cpython in the cases described by the tests. None of 2.5, 2.6 or 2.7 fall into this case, so anything that we might conceivably support that behaves this way can't be cpython.
> What do you suggest, should we rename `_checkCPythonWithEmptyPlatform` to `_checkPythonWithEmptyPlatform`?  And/or remove the `test_cpython` testcase?
   I'm suggesting that `_checkCPython` should return `False` in the fallthrough, instead of `True`, since now that we don't support <=2.4, that case can't be cpython. This could be done in a seperate ticket.
 3. ``
> I tried simplifying it but after several tries on the windows buildslave this is what worked.  I'd prefer to handle this in a different ticket, i dont have access to a local windows machine for proper testing etc
Sure.

Please merge, after after addressing 1, and filing tickets for 2+3.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_36"></a>|@thijstriemstra commented|
|-|-|

(In [39569]) address review comments, refs [#5387](https://github.com/twisted/twisted/issues/5387)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_37"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Replying to [tom.prince](#note_35):
>  1. `twisted/trial/runner.py`: `"%s" % (repr(x),)` is the same as `"%r" % (x,)`

Fixed.

>  3. ``
> > I tried simplifying it but after several tries on the windows buildslave this is what worked.  I'd prefer to handle this in a different ticket, i dont have access to a local windows machine for proper testing etc
> Sure.

Looks somebody opened a ticket for this already ([#6627](https://github.com/twisted/twisted/issues/6627)) that includes a patch that seems to have a better fix as well.

Forced a [buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-py24-refs-5387-4 build] and will merge when it's green (after I opened a ticket for 2).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_38"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [39703]) Merge remove-py24-refs-5387-4: Get rid of references and code specific to Python 2.4.

Author: thijs, bartekci                 
Reviewer: jesstess, MostAwesomeDude, jerith, ralphm, rockstar, tom.prince
Fixes: [#5387](https://github.com/twisted/twisted/issues/5387)
